prov/util/src/util_shm.c

```diff
@@ -145,6 +145,35 @@ size_t smr_calculate_size_offsets(size_t tx_count, size_t rx_count,
 	return total_size;
 }
 
+static int smr_retry_map(const char *name, int *fd)
+{
+	char tmp[NAME_MAX];
+	struct smr_region *old_shm;
+	struct stat sts;
+	int shm_pid;
+
+	*fd = shm_open(name, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+	if (*fd < 0)
+		return -errno;
+
+	old_shm = mmap(NULL, sizeof(*old_shm), PROT_READ | PROT_WRITE,
+		       MAP_SHARED, *fd, 0);
+	if (old_shm == MAP_FAILED)
+		return -errno;
+
+	shm_pid = old_shm->pid;
```

Comment:
I missed this the first time, but you'll want to add a version check to verify the pid offset location.

Reply:
The pid offset has always been the same so will be consistent across versions. Also, can we reasonable assume that the shm provider shouldn't have any version issues given that they will always be on the same node? (I know that it is possible to be using different versions but just wondering if how realistic/safe we are being with the provider in general when it comes to versioning)

Reply:
A single system can have multiple library versions installed, or a library could be statically linked.  Version mismatches could occur even on a single system.  The PID offset and size may have been the same for prior versions, but that doesn't mean it can't move in the future.  It basically just means that if the our version >= file version, then we can check the PID at the current offset/size.  If our version < file version, then any fields past the version cannot be relied on without the use of a time traveling DeLorean.